emmet-core/emmet/core/qchem/calculation.py

```diff
@@ -0,0 +1,437 @@
+        store_energy_trajectory
+            Whether to store the energy trajectory during a QChem calculation #TODO: Revisit this- False for now.
+        qcinput_kwargs
+            Additional keyword arguments that will be passed to the qcinput file
+        qcoutput_kwargs
+            Additional keyword arguments that will be passed to the qcoutput file
+
+        Returns
+        -------
+        Calculation
+            A QChem calculation document.
+        """
+
+        dir_name = Path(dir_name)
+        qcinput_file = dir_name / qcinput_file
+        qcoutput_file = dir_name / qcoutput_file
+
+        output_file_paths = _find_qchem_files(dir_name)
+
+        qcinput_kwargs = qcinput_kwargs if qcinput_kwargs else {}
+        qcinput = QCInput(qcinput_file, **qcinput_kwargs)
+
+        qcoutput_kwargs = qcoutput_kwargs if qcoutput_kwargs else {}
+        qcoutput = QCOutput(qcoutput_file, **qcoutput_kwargs)
+
+        completed_at = str(datetime.fromtimestamp(qcoutput_file.stat().st_mtime))
+
+        input_doc = CalculationInput.from_qcinput(qcinput)
+        output_doc = CalculationOutput.from_qcoutput(qcoutput)
+
+        has_qchem_completed = (
+            QChemStatus.SUCCESS
+            if qcoutput.data.get("completion", [])
+            else QChemStatus.FAILED
+        )
+
+        if store_energy_trajectory:
+            print("Still have to figure the energy trajectory")
+
+        return cls(
+            dir_name=str(dir_name),
+            task_name=task_name,
+            qchem_version=qcoutput.data["version"],
+            has_qchem_completed=has_qchem_completed,
+            completed_at=completed_at,
+            input=input_doc,
+            output=output_doc,
+            output_file_paths={k.name.lower(): v for k, v in output_file_paths.items()},
+            level_of_theory=level_of_theory(input_doc.rem),
+            task_type=task_type(input_doc.dict()),
+            calc_type=calc_type(input_doc.dict()),
+        )
+
+
+def _find_qchem_files(
+    path: Union[str, Path],
+) -> Dict[str, Any]:
+    """
+    Find QChem files in a directory.
+
+    Only the mol.qout file (or alternatively files
+    with the task name as an extension, e.g., mol.qout.opt_0.gz, mol.qout.freq_1.gz, or something like this...)
+    will be returned.
+
+    Parameters
+    ----------
+    path
+        Path to a directory to search.
+
+    Returns
+    -------
+    Dict[str, Any]
+        The filenames of the calculation outputs for each QChem task, given as a ordered dictionary of::
+
+            {
+                task_name:{
+                    "qchem_out_file": qcrun_filename,
+                },
+                ...
+            }
+    If there is only 1 qout file task_name will be "standard" otherwise it will be the extension name like "opt_0"
+    """
+    path = Path(path)
+    task_files = OrderedDict()
+    print(Path)
+
+    in_file_pattern = re.compile(r"^(?P<in_task_name>mol\.qin(?:\..+)?)\.gz$")
```

Comment:
Probably worth accepting *.inp and *.out, since those are actually Q-Chem's preferred input and output suffixes (Sam did his own thing).

Reply:
I've also seen `qcin` and `qcout`. Maybe the input and output extensions to search for should be kwargs? (list type, defaulting to just `[".qin"]`?